rotkehlchen/data_import/importers/bitcoin_tax.py

```diff
@@ -0,0 +1,337 @@
+            spend_asset = quote_asset
+            spend_amount = quote_asset_amount
+        elif action == 'SELL':
+            receive_asset = quote_asset
+            receive_amount = quote_asset_amount
+            spend_asset = base_asset
+            spend_amount = base_asset_amount
+        else:  # action = 'SWAP'
+            # The SWAP action is used when a crypto asset is renamed or forked.
+            # It is assumed that rotki knows if the new asset has been renamed or forked.
+            # We detect if it is a rename or a fork and handle the fork case with a
+            # new receive history event. In case of rename nothing needs to be done.
+            event_type = HistoryEventType.RECEIVE
+            try:
+                receive_asset = quote_asset.resolve_to_crypto_asset()
+            except UnknownAsset:
+                log.info(
+                    f'Ignoring unknown crypto asset {quote_asset} in BitcoinTax CSV SWAP action')  # noqa: E501
+                return  # skip this csv row
+
+            if receive_asset.forked is None:
+                log.info(
+                    f'Ignoring renamed crypto asset {receive_asset} in BitcoinTax CSV SWAP action')  # noqa: E501
+                return  # skip this csv row
+
+            receive_amount = quote_asset_amount
+            receive_event = HistoryEvent(
+                event_identifier=event_identifier,
+                sequence_index=0,
+                timestamp=timestamp,
+                location=location,
+                asset=receive_asset,
+                balance=Balance(
+                    amount=receive_amount,
+                    usd_value=ZERO,
+                ),
+                notes=memo,
+                event_type=event_type,
+                event_subtype=HistoryEventSubType.RECEIVE,
+            )
+            events.append(receive_event)
+            self.add_history_events(cursor, events)
+            return
+
+        spend_event = HistoryEvent(
+            event_identifier=event_identifier,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=location,
+            asset=spend_asset,
+            balance=Balance(
+                amount=spend_amount,
+                usd_value=ZERO,
+            ),
+            notes=memo,
+            event_type=event_type,
+            event_subtype=HistoryEventSubType.SPEND,
+        )
+        receive_event = HistoryEvent(
+            event_identifier=event_identifier,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=location,
+            asset=receive_asset,
+            balance=Balance(
+                amount=receive_amount,
+                usd_value=ZERO,
+            ),
+            notes=memo,
+            event_type=event_type,
+            event_subtype=HistoryEventSubType.RECEIVE,
+        )
+        events.extend([spend_event, receive_event])
+        if fee_amount != ZERO and fee_asset is not None:
```

Comment:
This probably needs some special accounting treatment to be defined in order to respect `should_calculate_fee` setting

Reply:
As I saw the `should_calculate_fee` is specific to the `Trade `dataclass while this one here is a HistoryEvent. But I'm not familiar with accounting so not sure about this.

Reply:
> should_calculate_fee is specific to the Trade

It should not be specific to `Trade` dataclass since a collection of history events (2 or 3 depending on fee) can represent exactly same action as `Trade` class represents. And we are actually moving right now from `Trade`, `LedgerAction`, etc. dataclasses towards storing everything in `HistoryEvent`s.

What I just realized though is that `HistoryEvent.process` works only with kraken staking events. And I don't know how/where the rest kraken events are processed. @yabirgb do you know?

Reply:
Yes, that method needs adjustment. In fact the imported events I believe they won't be correctly processed. We need more tests for accounting

Reply:
Created an issue for it https://github.com/rotki/rotki/issues/6162 and checking it